watcher.py

```diff
@@ -96,7 +125,7 @@ def run(self):
             self.results[test]['subtest'][subtest]['data'] = result
 
         self.log('Cleaning up recordings...')
-        for test, result in self.results.iteritems():
+        for test, result in list(self.results.iteritems()):
```

Comment:
I'd gone for `iteritems` only for the sake of having an iterator here. Why do we need a list? If we really need it, we could just have `self.results.items()`